core/graph/src/main/java/uk/gov/gchq/gaffer/graph/library/GraphLibrary.java

```diff
@@ -90,11 +90,25 @@ public StoreProperties getProperties(final String propertiesId) {
         return _getProperties(propertiesId);
     }
 
+    public boolean exists(final String graphId) {
+        if (getIds(graphId) != null) {
+            return true;
+        }
+        return false;
+    }
+
+    public void addSchema(final String schemaId, final Schema schema) throws OverwritingException {
+        if (null != schema) {
+            final byte[] schemaJson = schema.toJson(false);
+            addSchema(schemaId, schemaJson);
+        }
+    }
+
     protected abstract void _addIds(final String graphId, final Pair<String, String> schemaAndPropsIds) throws OverwritingException;
 
-    protected abstract void _addSchema(final String schemaId, final byte[] schema) throws OverwritingException;
+    public abstract void addSchema(final String schemaId, final byte[] schema) throws OverwritingException;
```

Comment:
We should expose an addSchema(String schemaId, Schema schema) method instead of this method that takes bytes.